### PR TITLE
Gh 340 project should build on ms windows

### PR DESCRIPTION
**Pull request by dev-958**
Changes to allow project to build on MS Windows, MS WSL Linux and Linux.

Note the addition of Hadoop binaries for windows.

### CODE REVIEW

File: service-impl/cache-service-impl/properties-backing-store/src/test/java/uk/gov/gchq/palisade/cache/service/impl/PropertiesBackingStoreTest.java [hunk 1]
```diff
@@ -33,15 +32,9 @@ public static void createTestStore() throws IOException {
         testFile = Files.createTempFile(null, null);
     }
 
-    @AfterClass
-    public static void removeTestStore() throws IOException {
-        Files.deleteIfExists(testFile);
-    }
-
```
[the remark this answers is not in the document]
Comment (dev-958) [reply service-impl/cache-service-impl/properties-backing-store/src/test/java/uk/gov/gchq/palisade/cache/service/impl/PropertiesBackingStoreTest.java]: On Windows attempting to delete this causes a security violation exception, one would expect temp files to be cleared by infrastructure.

File: example/example-model/src/main/java/uk/gov/gchq/palisade/example/util/ExampleFileUtil.java [hunk 2]
```diff
@@ -76,21 +78,31 @@ public static URI convertToFileURI(final String path) {
...
-
-        // if this path has the the file:// scheme, then convert it via URI
-        Path file;
-        if (FileSystems.getDefault().provider().getScheme().equals(uriPath.getScheme())) {
-            file = Paths.get(uriPath);
-        } else {
-            file = Paths.get(path);
+            try {
+                uriPath = Paths.get(path).normalize().toUri();
+            } catch (Throwable f) {
```
[the remark this answers is not in the document]
Comment (dev-958) [reply example/example-model/src/main/java/uk/gov/gchq/palisade/example/util/ExampleFileUtil.java]: sure.

File: example/example-model/src/test/java/uk/gov/gchq/palisade/example/client/ExampleFileUtilTest.java [hunk 3]
```diff
@@ -147,7 +149,8 @@ public void shouldNotChangeOtherProtocol() {
     @Test
     public void shouldNotChangeNonExistentFile() {
         //Given
-        String expected = "file:///nowhere/no_file";
+        //String expected = "file:///nowhere/no_file";
```
[the remark this answers is not in the document]
Comment (dev-958) [reply example/example-model/src/test/java/uk/gov/gchq/palisade/example/client/ExampleFileUtilTest.java]: yup.

File: service-impl/resource-service-impl/hadoop-resource-service/pom.xml [hunk 4]
```diff
@@ -118,6 +118,30 @@
...
+            <artifactId>hadoop-mapreduce-client-common</artifactId>
+            <version>${hadoop.version}</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-minicluster</artifactId>
+            <version>3.2.0</version>
+            <scope>test</scope>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (dev-958) [reply service-impl/resource-service-impl/hadoop-resource-service/pom.xml]: there are some annoying transitive dependencies required for the hadoop test utils, only required to load under windows though.

File: service-impl/cache-service-impl/properties-backing-store/src/test/java/uk/gov/gchq/palisade/cache/service/impl/PropertiesBackingStoreTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (m09526) [reply service-impl/cache-service-impl/properties-backing-store/src/test/java/uk/gov/gchq/palisade/cache/service/impl/PropertiesBackingStoreTest.java]: agreed

File: service-impl/resource-service-impl/hadoop-resource-service/pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (m09526) [reply service-impl/resource-service-impl/hadoop-resource-service/pom.xml]: okay, we'll leave these in, but I'm mindful about the amount of Palisade which is actually just hadoop dependencies. Many of the libraries we bring in are actually just things referenced by hadoop.

**Pull request merged**